examples/platform/linux/AppMain.cpp

```diff
@@ -369,6 +369,71 @@ class SampleTestEventTriggerHandler : public TestEventTriggerHandler
     }
 };
 
+#if CHIP_DEVICE_CONFIG_ENABLE_WIFIPAF
+void int_array_add_unique(uint16_t ** res, uint16_t a)
+{
+    size_t reslen, max_size;
+    uint16_t * n;
+
+    for (reslen = 0; *res && (*res)[reslen]; reslen++)
+    {
+        if ((*res)[reslen] == a)
+            return; /* already in the list */
+    }
+    max_size = (size_t) -1;
+    if (reslen > max_size - 2)
+    {
+        /* This should not really happen in practice, but if it did,
+         * something would overflow. Do not try to add the new value;
+         * instead, make this behave like memory allocation failure to
+         * avoid messing up memory. */
+        free(*res);
+        *res = NULL;
+        return;
+    }
+    n = (uint16_t *) realloc(*res, (reslen + 2) * sizeof(uint16_t));
+    if (n == NULL)
+    {
+        free(*res);
+        *res = NULL;
+        return;
+    }
+    n[reslen]     = a;
+    n[reslen + 1] = 0;
+    *res          = n;
+}
+
+static uint16_t WiFiPAFGet_FreqList(char * ArgStrn, uint16_t ** pfreq_list)
```

Comment:
Please also improve readability on this method to:
  - describe what it does and how (what do the arguments mean, how are they transformed, usage examples
  - rename some variables (what is `len` for example)?
  - is `freq_list` needed? we seem to use it just for a get and to later place it in pfreq_list. If needed this should be moved to decrease its scope.
  - Can we somehow make it so that ArgStrn can be const? generally this seems the intent and we use non-const so that strtok works.

Reply:
I will change it